server/lib/resolvers/resolverMapper.js

```diff
@@ -53,3 +72,36 @@ module.exports = function (dataSources, resolverMappings) {
 
   return resolvers
 }
+
+function resolveAndPublish (resolverFn, publishOpts) {
+  return (obj, args, context, info) => {
+    return new Promise(async (resolve, reject) => {
+      try {
+        const result = await resolverFn(obj, args, context, info)
+        resolve(result)
+
+        const { pubsub, topic, compiledPayload } = publishOpts
```

Comment:
I would wrap the code from here into separate function like `notifySubscriptions()` that could be used in different context.